[[source]]
name = "pypi"
url = "https://pypi.org/simple"
verify_ssl = true

[packages]
dlinfo = {editable = true,path = "."}

[dev-packages]
pylint = "*"
pytest = "*"
pytest-cov = "*"

# python3.7 compatibility
# > File "[...]/python3.7/site-packages/_pytest/compat.py", line 56, in <module>
# >  import importlib_metadata  # noqa: F401
importlib-metadata = {markers = "python_version < '3.8'"}
# https://github.com/PyCQA/isort/commit/47cedf3583f338e8ce9c906fc95c9fee6c57404d
isort = "<5.12"
# > File "[...]/python3.7/site-packages/importlib_metadata/__init__.py", line 6, in <module>
# >  import zipp
zipp = {markers = "python_version < '3.8'"}

# >    File "[...]/lib/python3.10/site-packages/_pytest/_code/code.py", line 60, in <module>
# >     from exceptiongroup import BaseExceptionGroup
# > ModuleNotFoundError: No module named 'exceptiongroup'
exceptiongroup = {markers = "python_version < '3.11'"}
# >   File "[...]/lib/python3.10/site-packages/_pytest/config/findpaths.py", line 71, in load_config_dict_from_file
# >     import tomli as tomllib
# > ModuleNotFoundError: No module named 'tomli'
tomli = {markers = "python_version < '3.11'"}
# >   File "[...]/python3.9/site-packages/astroid/decorators.py", line 25, in <module>
# >     from typing_extensions import ParamSpec
# > ModuleNotFoundError: No module named 'typing_extensions'
typing-extensions = "*"
# >   File "[...]/lib/python3.10/site-packages/astroid/decorators.py", line 16, in <module>
# >     import wrapt
# > ModuleNotFoundError: No module named 'wrapt'
wrapt = "*"
# remove `"markers": "python_version >= '3.11'"` to workaround:
# >   File "[...]/lib/python3.7/site-packages/pylint/lint/parallel.py", line 13, in <module>
# >     import dill
# > ModuleNotFoundError: No module named 'dill'
dill = "*"

[requires]
python_version = "3"

# Pipfile syntax: https://github.com/pypa/pipfile#pipfile
